engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldPreGenerationScreen.java

```diff
@@ -15,31 +15,253 @@
+                    updatePreview();
+                } catch (UnresolvedWorldGeneratorException e) {
+                    e.getMessage();
+                }
+            }
+        });
+
+        WidgetUtil.trySubscribe(this, "reRoll", button -> {
+            worldGenerator.setWorldSeed(createSeed(selectedWorld));
+            getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Ready to roll!", "World seed for " + selectedWorld + " has been changed");
+        });
+
         StartPlayingScreen startPlayingScreen = getManager().createScreen(StartPlayingScreen.ASSET_URI, StartPlayingScreen.class);
-        WidgetUtil.trySubscribe(this, "continue", button ->
-                triggerForwardAnimation(startPlayingScreen)
-        );
+        WidgetUtil.trySubscribe(this, "continue", button -> {
+            startPlayingScreen.setTargetWorld(findWorldByName(), texture, context);
+            triggerForwardAnimation(startPlayingScreen);
+        });
 
         WorldSetupScreen worldSetupScreen = getManager().createScreen(WorldSetupScreen.ASSET_URI, WorldSetupScreen.class);
-
-        WidgetUtil.trySubscribe(this, "config", button ->
-                triggerForwardAnimation(worldSetupScreen)
-        );
+        WidgetUtil.trySubscribe(this, "config", button -> {
+            try {
+                if (!selectedWorld.isEmpty()) {
+                    worldSetupScreen.setWorld(context, findWorldByName());
+                    triggerForwardAnimation(worldSetupScreen);
+                } else {
+                    getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Worlds List Empty!", "No world found to configure.");
+                }
+            } catch (UnresolvedWorldGeneratorException e) {
+                e.getMessage();
+            }
+        });
 
         WidgetUtil.trySubscribe(this, "close", button ->
                 triggerBackAnimation()
         );
     }
+
+    @Override
+    public void onOpened() {
+        try {
+            if (findWorldByName().getWorldGenerator() == null) {
+                worldGenerator = WorldGeneratorManager.createWorldGenerator(findWorldByName().getWorldGeneratorInfo().getUri(), context, environment);
+                findWorldByName().setWorldGenerator(worldGenerator);
+            } else {
+                worldGenerator = findWorldByName().getWorldGenerator();
+            }
+            if (worldGenerator.getWorldSeed().isEmpty()) {
+                worldGenerator.setWorldSeed(createSeed(selectedWorld));
+            }
+            previewGen = new FacetLayerPreview(environment, worldGenerator);
+            updatePreview();
+        } catch (UnresolvedWorldGeneratorException e) {
+            e.getMessage();
+        }
+    }
+
+    /**
+     * Generates a texture and set's to the image view, thus previewing the world.
+     */
+    private void genTexture() {
+        int imgWidth = 384;
```

Comment:
Should the size be variable in case the user resizes the window? If not maybe extract to a constant or two.

Resizing support could be added later, not important right now

Reply:
set's -> sets
..sets *it* to..

You really need to be careful with your apostrophes.